src/testdir/test_termdebug.vim

```diff
@@ -300,4 +300,41 @@ func Test_termdebug_mapping()
   %bw!
 endfunc
 
+func Test_termdebug_bufnames()
+  " Test if user has filename/folders named gdb, Termdebug-gdb-console,
+  " etc. in the current directory
+  let g:termdebug_config = {}
+  let g:termdebug_config['use_prompt'] = 1
+  let filename = 'gdb'
+  let replacement_filename = 'Termdebug-gdb-console'
+
+  call writefile(['This', 'is', 'a', 'test'], filename, 'D')
+  " Throw away the file once the test has done.
+  Termdebug
+  sleep 2
```

Comment:
Using hard-coded sleeps in tests will increase the time it takes to run the tests.  Can you change these
to WaitForAssert() calls?

Reply:
Ok, I replaced the `sleep 2` with `  call WaitForAssert({-> assert_equal(3, winnr('$'))}` , which, in my understanding, wait until the third window is available (in this case, given the termdebug setup, it means that termdebug is up and running).